src/enumerable.cr

```diff
@@ -609,6 +609,37 @@ module Enumerable(T)
     memo
   end
 
+  # Similar to `reduce`, but instead of raising when the input is empty,
+  # return `nil`
+  #
+  # ```
+  # ([] of Int32).reduce? { |acc, i| acc + i } # => nil
+  # ```
+  def reduce?
+    memo = uninitialized T
+    found = false
+
+    each do |elem|
+      memo = found ? (yield memo, elem) : elem
+      found = true
+    end
+
+    found ? memo : nil
+  end
+
+  # Similar to `reduce?`, however instead of returning `nil` when the input is empty,
+  # return the initial value of the accumulator.
```

Comment:
This is incorrect, it's exactly the same method implementation as `reduce` above. Actually the original impl of `reduce(memo)` is incorrect and does not raise `Enumerable::EmptyError` at all.

Reply:
`reduce?(memo)` should be removed since it's identical to `reduce(memo)`.